gammapy/estimators/flux_map.py

```diff
@@ -0,0 +1,480 @@
+        norm = self.norm.get_by_coord(coords) * self.norm.unit
+        norm_err, norm_errn, norm_errp, norm_ul = None, None, None, None
+        norm_scan, stat_scan = None, None
+        if self.norm_err is not None:
+            norm_err = self.norm_err.get_by_coord(coords) * self.norm_err.unit
+        if self.norm_errn is not None:
+            norm_errn = (
+                self.norm_errn.get_by_coord(coords) * self.norm_errn.unit
+            )
+        if self.norm_errp is not None:
+            norm_errp = (
+                self.norm_errp.get_by_coord(coords) * self.norm_errp.unit
+            )
+        if self.norm_ul is not None:
+            norm_ul = (
+                self.norm_ul.get_by_coord(coords) * self.norm_ul.unit
+            )
+        # TODO: add support of norm and stat scan
+
+        rows = []
+        for idx, energy in enumerate(self.e_ref):
+            result = dict()
+            result["e_ref"] = energy
+            result["e_min"] = self.e_min[idx]
+            result["e_max"] = self.e_max[idx]
+            result["ref_dnde"] = ref[idx]
+            result["norm"] = norm[idx]
+            if norm_err is not None:
+                result["norm_err"] = norm_err[idx]
+            if norm_errn is not None:
+                result["norm_errn"] = norm_errn[idx]
+            if norm_errp is not None:
+                result["norm_errp"] = norm_errp[idx]
+            if norm_ul is not None:
+                result["norm_ul"] = norm_ul[idx]
+            if norm_scan is not None:
+                result["norm_scan"] = norm_scan[idx]
+                result["stat_scan"] = stat_scan[idx]
+            rows.append(result)
+        table = table_from_row_data(rows=rows, meta={"SED_TYPE": "likelihood"})
+        return FluxPoints(table)
+
+    def to_dict(self, sed_type="likelihood"):
+        """Return maps in a given SED type in the form of a dictionary.
+
+        Parameters
+        ----------
+        sed_type : str
+            sed type to convert to. Default is `Likelihood`
+
+        Returns
+        -------
+        map_dict : dict
+            dictionary containing the requested maps.
+        """
+        result = {}
+        for entry in REQUIRED_MAPS[sed_type]:
+            result[entry] = self.__getattribute__(entry)
+
+        for entry in OPTIONAL_MAPS[sed_type]:
+            res = self.__getattribute__(entry)
+            if res is not None:
+                result[entry] = res
+
+        return result
+
+    def write(self, filename, overwrite=False, sed_type="likelihood"):
+        """Write flux map to file.
+
+        Parameters
+        ----------
+        filename : str
+            Filename to write to.
+        overwrite : bool
+            Overwrite file if it exists.
+        sed_type : str
+            sed type to convert to. Default is `Lielihood`
+        """
+        self.to_hdulist().writeto(str(make_path(filename)), overwrite=overwrite)
```

Comment:
To reduce the file size, one could remove the `BANDS` HDU here...

Reply:
Indeed, but there is currently a bug in `Maps` preventing that I think.